Formula/hdf5.rb

```diff
@@ -58,7 +58,8 @@ def install
     system "./configure", *args
 
     # Avoid shims in settings file
-    inreplace "src/libhdf5.settings", HOMEBREW_LIBRARY/"Homebrew/shims/mac/super/clang", "/usr/bin/clang"
+    inreplace "src/libhdf5.settings", %r{#{HOMEBREW_SHIMS_PATH}/[^/]+/super/#{ENV.cc}}, ENV.cc
+    on_linux { inreplace "src/libhdf5.settings", %r{#{HOMEBREW_SHIMS_PATH}/[^/]+/super/#{ENV.cxx}}, ENV.cxx }
```

Comment:
Probably need to escape characters like `+` inside `g++`

Reply:
Yeah that appears to be necessary here.  We got lucky on macOS because you `clang` is just a substring of `clang++`, so one replacement handles both.  Not so for GCC unfortunately.

Reply:
I guess it is also possible to aggressively deleted all shim paths,
- `%r{#{HOMEBREW_SHIMS_PATH}/[^/]+/super/, ""`

Or some slightly restricted:
- `%r{#{HOMEBREW_SHIMS_PATH}/[^/]+/super/(c|g), "\\1"`
- `%r{#{HOMEBREW_SHIMS_PATH}/[^/]+/super/(clang|gcc|g\+\+), "\\1"`

Though, there are demerits/risks with these alternatives as opposed to:
- `%r{#{HOMEBREW_SHIMS_PATH}/[^/]+/super/#{Regexp.escape(ENV.cxx)}}, ENV.cxx`

Reply:
I used that last one and it worked great.  When CI finishes this should be good to go.